[2024-06-14T08:25:05.475+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-14T08:25:05.508+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparking_flow.python_job scheduled__2024-06-13T00:00:00+00:00 [queued]>
[2024-06-14T08:25:05.516+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparking_flow.python_job scheduled__2024-06-13T00:00:00+00:00 [queued]>
[2024-06-14T08:25:05.518+0000] {taskinstance.py:2306} INFO - Starting attempt 7 of 7
[2024-06-14T08:25:05.532+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): python_job> on 2024-06-13 00:00:00+00:00
[2024-06-14T08:25:05.539+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sparking_flow', 'python_job', 'scheduled__2024-06-13T00:00:00+00:00', '--job-id', '9', '--raw', '--subdir', 'DAGS_FOLDER/spark_***.py', '--cfg-path', '/tmp/tmpl7b7yfhy']
[2024-06-14T08:25:05.542+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1819) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-06-14T08:25:05.542+0000] {standard_task_runner.py:91} INFO - Job 9: Subtask python_job
[2024-06-14T08:25:05.543+0000] {standard_task_runner.py:63} INFO - Started process 1820 to run task
[2024-06-14T08:25:05.595+0000] {task_command.py:426} INFO - Running <TaskInstance: sparking_flow.python_job scheduled__2024-06-13T00:00:00+00:00 [running]> on host e342e0ac560c
[2024-06-14T08:25:05.677+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Selva' AIRFLOW_CTX_DAG_ID='sparking_flow' AIRFLOW_CTX_TASK_ID='python_job' AIRFLOW_CTX_EXECUTION_DATE='2024-06-13T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='7' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-13T00:00:00+00:00'
[2024-06-14T08:25:05.678+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-14T08:25:05.698+0000] {base.py:84} INFO - Using connection ID 'spark-conn' for task execution.
[2024-06-14T08:25:05.700+0000] {spark_submit.py:401} INFO - Spark-Submit cmd: spark3-submit --master spark://spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py
[2024-06-14T08:25:05.704+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-14T08:25:05.705+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 481, in submit
    self._submit_sp = subprocess.Popen(
                      ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.12/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 13] Permission denied: 'spark3-submit'
[2024-06-14T08:25:05.717+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=sparking_flow, task_id=python_job, run_id=scheduled__2024-06-13T00:00:00+00:00, execution_date=20240613T000000, start_date=20240614T082505, end_date=20240614T082505
[2024-06-14T08:25:05.731+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 9 for task python_job ([Errno 13] Permission denied: 'spark3-submit'; 1820)
[2024-06-14T08:25:05.759+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-14T08:25:05.780+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-14T08:25:05.786+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
